include/sys/uzfs_zvol.h

```diff
@@ -149,11 +147,12 @@ typedef struct uzfs_zvol_blk_phy {
 typedef struct uzfs_io_chunk_list {
 	uint64_t offset;
 	uint64_t len;
+	uint64_t io_number;
 	char *buf;
 	list_node_t link;
 } uzfs_io_chunk_list_t;
 
-typedef int (uzfs_txg_diff_traverse_cb_t)(off_t offset, size_t len,
-    uint64_t blkid, void *arg);
+typedef int (uzfs_get_io_diff_cb_t)(off_t offset, size_t len,
+    blk_metadata_t *metadata, objset_t *obj, void *arg);
```

Comment:
how about passing zvol_state_t and using uzfs_read_data rather than dmu_read in callback implementation?